smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -298,51 +298,49 @@ public static void writeRetryableTrait(TypeScriptWriter writer, StructureShape e
      * returned.
      *
      * @param context The generation context.
-     * @param operation The operation to generate for.
      * @param responseType The response type for the HTTP protocol.
      * @param errorCodeGenerator A consumer
      * @param shouldParseErrorBody Flag indicating whether need to parse response body in this dispatcher function
      * @param bodyErrorLocationModifier A function that returns the location of an error in a body given a data source.
      * @param operationErrorsToShapes A map of error names to their {@link ShapeId}.
      * @return A set of all error structure shapes for the operation that were dispatched to.
      */
-    static Set<StructureShape> generateErrorDispatcher(
-            GenerationContext context,
-            OperationShape operation,
-            SymbolReference responseType,
-            Consumer<GenerationContext> errorCodeGenerator,
-            boolean shouldParseErrorBody,
-            BiFunction<GenerationContext, String, String> bodyErrorLocationModifier,
-            BiFunction<GenerationContext, OperationShape, Map<String, ShapeId>> operationErrorsToShapes
+    static Set<StructureShape> generateUnifiedErrorDispatcher(
+        GenerationContext context,
+        List<OperationShape> operations,
+        SymbolReference responseType,
+        Consumer<GenerationContext> errorCodeGenerator,
+        boolean shouldParseErrorBody,
+        BiFunction<GenerationContext, String, String> bodyErrorLocationModifier,
+        BiFunction<GenerationContext, List<OperationShape>, Map<String, ShapeId>> operationErrorsToShapes
     ) {
         TypeScriptWriter writer = context.getWriter();
         SymbolProvider symbolProvider = context.getSymbolProvider();
         Set<StructureShape> errorShapes = new TreeSet<>();
 
-        Symbol symbol = symbolProvider.toSymbol(operation);
-        Symbol outputType = symbol.expectProperty("outputType", Symbol.class);
-        String errorMethodName = ProtocolGenerator.getDeserFunctionShortName(symbol) + "Error";
-        String errorMethodLongName = ProtocolGenerator.getDeserFunctionName(symbol, context.getProtocolName())
-                + "Error";
+        String errorMethodName = "de_CommandError";
+        String errorMethodLongName = "deserialize_"
+            + ProtocolGenerator.getSanitizedName(context.getProtocolName())
+            + "CommandError";
 
         writer.writeDocs(errorMethodLongName);
-                writer.openBlock("const $L = async(\n"
-                               + "  output: $T,\n"
-                               + "  context: __SerdeContext,\n"
-                               + "): Promise<$T> => {", "}", errorMethodName, responseType, outputType, () -> {
+        writer.openBlock("const $L = async(\n"
+            + "  output: $T,\n"
+            + "  context: __SerdeContext,\n"
+            + "): Promise<unknown> => {", "}", errorMethodName, responseType, () -> {
```

Comment:
Clarifying question: is the change to `unknown` a breaking change, or does it not matter?

Reply:
it does not matter, it should be `Promise<never>` since this function is supposed to throw all the time